spring-resttemplate-2/src/main/java/com/baeldung/resttemplate/json/provider/controller/UserController.java

```diff
@@ -0,0 +1,23 @@
+package com.baeldung.resttemplate.json.provider.controller;
+
+import com.baeldung.resttemplate.json.model.User;
+import com.baeldung.resttemplate.json.service.UserService;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+    @RequestMapping(value = "/users", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public @ResponseBody
+    List<User> getUsers() {
```

Comment:
Is this an essential part of the example? Are you not mocking it out in the tests/example code.

We're only interested in the consumer of this API, not the producer.

Reply:
Thank you! I realise this is not needed now.